### PR TITLE
fix: Support length on slices with null

### LINKED ISSUE #744: length on slices with null doesn't work
**Describe the bug**

`length` kernel (for string length) doesn't work on sliced string arrays containing null.

**To Reproduce**

Modify the existing tests to include null:

```
    /// Tests with an offset
    #[test]
    fn bit_length_offsets() -> Result<()> {
        let a = StringArray::from(vec![Some("hello"), Some(" "), Some("world"), None]);
        let b = a.slice(1, 3);
        let result = bit_length(b.as_ref())?;
        let result: &Int32Array = as_primitive_array(&result);

        let expected = Int32Array::from(vec![Some(8), Some(40), None]);
        assert_eq!(&expected, result);

        Ok(())
    }
```

**Expected behavior**
Test passes.

**Actual behavior**
Test does not pass.

### PR DESCRIPTION
**Pull request by bjchambers**
# Which issue does this PR close?

Closes #744.

# Rationale for this change
 
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/src/compute/kernels/length.rs [hunk 1]
```diff
@@ -243,18 +242,13 @@ mod tests {
...
-                .buffers(a.data_ref().buffers().to_vec())
-                .build(),
-        );
-        let result = length(b.as_ref())?;
-
-        let expected = Int32Array::from(vec![1, 5]);
-        assert_eq!(expected.data(), result.data());
+        let a = StringArray::from(vec![Some("hello"), Some(" "), Some("world"), None]);
+        let b = a.slice(1, 3);
+        let result = bit_length(b.as_ref())?;
```
Comment (matthewmturner) [inline arrow/src/compute/kernels/length.rs:247]: I believe this should use `length` instead of `bit_length`

Comment (alamb) [reply arrow/src/compute/kernels/length.rs:247]: good eyes. I agree that the intent of this test is to test `length` and thus it should not be calling `bit_length`

**Review by alamb — COMMENTED** [review]
Thank you @bjchambers  -- this looks great other than the issue with the test identified by @matthewmturner

File: arrow/src/compute/kernels/length.rs (hunk 1 above)
Comment (bjchambers) [reply arrow/src/compute/kernels/length.rs]: Fixed.

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @bjchambers !

**Pull request merged**